### PR TITLE
CNV-28335: Connecting top consumer cards to persistent user settings

### PR DESCRIPTION
**Pull request by metalice**
## 📝 Description

connection overview -> top consumers

All charts settings are now connecting to kubevirt user settings and will be persistent between logins in different sessions and not only over sessions.

## 🎥 Demo

> Please add a video or an image of the behavior/changes

### CODE REVIEW

File: src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts [hunk 1]
```diff
@@ -25,20 +26,20 @@ const useKubevirtUserSettings: UseKubevirtUserSettings = (key) => {
   const [error, setError] = useState<Error>();
   const [userName, setUserName] = useState<string>();
   const [userSettings, setUserSettings] = useState<UserSettingsState>();
-  const [loading, setLoading] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(true);
```
Comment (avivtur) [inline src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts:29]: is this loading meant for the userSettings? if yes, maybe rename it to clarify what is this loading for.

File: src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts [hunk 2]
```diff
@@ -25,20 +26,20 @@ const useKubevirtUserSettings: UseKubevirtUserSettings = (key) => {
...
 
   useEffect(() => {
     const fetchUserName = async () => {
       setError(null);
-      setLoading(true);
       try {
         const user = await k8sGet({ model: UserModel, path: '~' });
         setUserName(user?.metadata?.name?.replace(':', '-'));
       } catch (e) {
         setError(e);
```
Comment (avivtur) [inline src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts:40]: do we also want to set loading to false when we get an error?

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumerCard.tsx [hunk 3]
```diff
@@ -1,38 +1,59 @@
...
 import './TopConsumerCard.scss';
 
 type TopConsumersMetricCard = {
   cardID: string;
+  localStorageData: TopConsumersData;
+  setLocalStorageData: SetTopConsumerData;
 };
 
-const TopConsumerCard: React.FC<TopConsumersMetricCard> = ({ cardID }) => {
+const TopConsumerCard: React.FC<TopConsumersMetricCard> = ({
```
Comment (avivtur) [inline src/views/clusteroverview/TopConsumersTab/utils/TopConsumerCard.tsx:23]: can you please drop `React.` ?

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumersChartList.tsx [hunk 4]
```diff
@@ -22,13 +22,24 @@ import './TopConsumersChartList.scss';
 type TopConsumersChartListProps = {
   metric: TopConsumerMetric;
   scope: TopConsumerScope;
+  localStorageData: TopConsumersData;
 };
 
-export const TopConsumersChartList: React.FC<TopConsumersChartListProps> = ({ metric, scope }) => {
+export const TopConsumersChartList: React.FC<TopConsumersChartListProps> = ({
```
Comment (avivtur) [inline src/views/clusteroverview/TopConsumersTab/utils/TopConsumersChartList.tsx:28]: here too :)

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumersGridRow.tsx [hunk 5]
```diff
@@ -24,13 +32,25 @@ const TopConsumersGridRow: React.FC<TopConsumersGridRowProps> = ({
...
+          cardID={getTopConsumerCardID(rowNumber, 2)}
+        />
       </GridItem>
       <GridItem span={4} className="kv-top-consumers-card__card-grid-item">
-        <TopConsumerCard cardID={getTopConsumerCardID(rowNumber, 3)} />
+        <TopConsumerCard
+          localStorageData={localStorageData}
+          setLocalStorageData={setLocalStorageData}
+          cardID={getTopConsumerCardID(rowNumber, 3)}
+        />
```
Comment (avivtur) [inline src/views/clusteroverview/TopConsumersTab/utils/TopConsumersGridRow.tsx:53]: maybe map here?

File: src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts (hunk 2 above)
Comment (metalice) [reply src/utils/hooks/useKubevirtUserSettings/useKubevirtUserSettings.ts:40]: yes, loading isn't relevant to error, if u get an error loading is done, so its ok

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumersChartList.tsx (hunk 4 above)
Comment (metalice) [reply src/views/clusteroverview/TopConsumersTab/utils/TopConsumersChartList.tsx:28]: sure

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumerCard.tsx (hunk 3 above)
Comment (metalice) [reply src/views/clusteroverview/TopConsumersTab/utils/TopConsumerCard.tsx:23]: sure

File: src/views/clusteroverview/TopConsumersTab/utils/TopConsumersGridRow.tsx (hunk 5 above)
Comment (metalice) [reply src/views/clusteroverview/TopConsumersTab/utils/TopConsumersGridRow.tsx]: sure, added a map

**Review by avivtur — APPROVED** [review]

**Pull request merged**